src/storage/src/sink/kafka.rs

```diff
@@ -373,7 +373,7 @@ struct KafkaSinkState {
 
     progress_topic: String,
     progress_key: String,
-    progress_client: Arc<BaseConsumer<BrokerRewritingClientContext<MzClientContext>>>,
```

Comment:
I'm starting to think it might be a smell that we have this general-purpose `MzClientContext`. Makes it easy to just set it and forget, but then errors don't bubble up to users and just show up in logs.

Nothing to fix in this PR, though. Or maybe not at all... 🤷

Reply:
Yeah, interesting, unsure! It's true we never use the raw MzClientContext now as far as I know. But I feel like this stuff is still pretty in flux...